### PR TITLE
Drop 24.5 support 

### PR DESCRIPTION
**Pull request by progfolio**
See: #635

### CODE REVIEW

File: straight.el [hunk 1]
```diff
@@ -5084,14 +5026,7 @@ The default value is \"Processing\"."
...
                                         (setq next-repos (cdr next-repos))
                                         (cl-return-from loop))
-                                    (error e)
-                                    ;; Emacs 24.5 has a bug where the
-                                    ;; byte-compiler signals an unused
-                                    ;; argument warning for the target
-                                    ;; of a `condition-case' unless
-                                    ;; it's used on every error
-                                    ;; handler.
-                                    (quit (ignore e)))))
```
Comment (progfolio) [inline straight.el:5094]: Not entirely sure about this change

File: straight.el [hunk 2]
```diff
@@ -1230,8 +1185,8 @@ appropriately."
     (setq straight-find-flavor (straight--determine-find-flavor)))
   (memq symbol
         (pcase straight-find-flavor
-          (`gnu/bsd '(newermt))
-          (`busybox nil)
+          ('gnu/bsd '(newermt))
+          ('busybox nil)
```
Comment (raxod502) [inline straight.el:1189]: Nice, I'll have to remember to stop using the backticks now that we don't have to support that ancient version!

File: straight.el (hunk 1 above)
Comment (raxod502) [reply straight.el:5094]: I think we want to replace `(quit (ignore e))` with just `(quit)`.

File: straight.el (hunk 2 above)
Comment (progfolio) [reply straight.el:1189]: Yeah, that's one that I always forget about myself. Good to have to only remember one way now.

File: straight.el (hunk 1 above)
Comment (progfolio) [reply straight.el:5101]: Addressed in latest commit.

**Pull request merged**